# 04 — Ipotesi 3: scomposizione short/long (term structure)

MA breve (21g) vs MA lunga (126g) della vol: BEAR se la componente breve è sopra la lunga. È l'ipotesi più vicina a "disentangling vol into long/short".

In [1]:
import warnings; warnings.filterwarnings("ignore")
import numpy as np, pandas as pd
import vol_regime as vr
pd.set_option("display.float_format", lambda x: f"{x:.3f}")

vol = vr.load_vol()

## Hit ratio: il segnale predice la direzione del mese successivo? (vs proxy)

In [2]:
for ac, vcol in vr.VOL_FOR_CLASS.items():
    v = vol[vcol].dropna()
    sig = vr.signal_term_structure(v, 21, 126)
    print(f"==== {ac} (vol={vcol}) ====")
    print(vr.compare_table(sig, ac).to_string(float_format=lambda x: f"{x:.3f}"))
    print()

==== Equity (vol=VIX) ====
                hit_ratio       n  base_rate_up  bull_share
Equity (h=21d)                                             
signal              0.543 317.000         0.618       0.609
proxy               0.615 317.000         0.618       0.700

==== Fixed Income (vol=MOVE) ====
                      hit_ratio       n  base_rate_up  bull_share
Fixed Income (h=21d)                                             
signal                    0.513 279.000         0.505       0.599
proxy                     0.510 286.000         0.514       0.493



## Stabilità: hit ratio per anno (Equity)

In [3]:
v = vol["VIX"].dropna()
sig = vr.signal_term_structure(v, 21, 126)
px = vr.load_prices()["SPY"].dropna()
sig_al = sig.reindex(px.index).ffill()
vr.hit_ratio_by_year(sig_al, px).round(3)

2000   0.500
2001   0.083
2002   0.417
2003   0.667
2004   0.750
2005   0.667
2006   0.750
2007   0.417
2008   0.333
2009   0.833
2010   0.750
2011   0.333
2012   0.500
2013   0.500
2014   0.417
2015   0.417
2016   0.500
2017   0.667
2018   0.583
2019   0.583
2020   0.667
2021   0.750
2022   0.500
2023   0.583
2024   0.500
2025   0.583
2026   0.200
Name: correct, dtype: float64

**Conclusione (04).** Anche la scomposizione short/long con i dati gratuiti
**non batte il proxy** (≈0.54 equity). La vera term structure (futures VIX,
superfici di IV) richiederebbe Bloomberg — fuori scope ora.